code-style/scss/README.md

```diff
@@ -1,52 +1,257 @@
+
+#### Tools
+
+This section is specifically for preprocessor global mixins and functions. No CSS should be generated by the preprocessor from partials in this section. Some of our favorite global mixins to use are [unbuttonize](https://gist.github.com/zastrow/bb43d2b256d52738e0ffdee7f825e79f) to remove default button styles and [delist](https://gist.github.com/zastrow/94d97dfe2e7e2808bcf4dc45268e6338) to remove default list styles from `<ol>` and `<ul>`.
+
+#### Generic
+
+Here we define any generic styles that are not specific to the styles of the site. This section should include any Reset or Normalize libraries and any preferential base styles for elements. There shouldn’t be any classes or ids used in this section.
+
+#### Elements
+
+This is where we define styling for bare HTML elements (like H1, A, etc.). These elements come with default styling from the browser, so we can redefine them here.
+
+#### Objects
+
+Objects are class-based selectors that define undecorated design patterns, for example [the media object](http://www.stubbornella.org/content/2010/06/25/the-media-object-saves-hundreds-of-lines-of-code/) (a pattern with an image on the left and descriptive content on the right, like a Tweet for example) from OOCSS.
+
+#### Components
+
+This section is where we define our specific UI components. This is where the majority of our work takes place, and our UI components are often composed of Objects and Components.
+
+#### Utilities
+
+Utilities are helper classes with the ability to override anything that goes before it in the triangle. Utility classes are very specific. Sometimes we use utility classes to set up spacing systems, which would consist of creating classes for certain sizes of padding or margin to be used anywhere on the site, or to establish a hide helper class that our JavaScript can interact with to toggle the visibility of elements.
+
+---
+
+## Namespacing
+
+We namespace to articulate what class names we wrote. As with the [class naming section above](#Class-Naming-Theory), namespacing is a useful tool to describe what kind of class this is and how it functions. Also, like class naming, the more information in the namespace, the easier it is to comprehend for future developers.
+
+A namespace title should consist of more than one character but no more than four characters. One character rarely provides enough information to describe the type of class, and more than four characters cause the namespace to compete with the class name. Namespaces should be quick to comprehend but only act as an aid to define the type of styles that will exist and how they function.
+
+Our namespacing is derived from the ITCSS section the class belongs in. Since settings, tools, generic, and elements are only used in file naming (and usually use the whole title in the file name), we have only defined ITCSS prefixes for the sections used for class naming:
+
+- `.obj-` (objects)
+- `.cmp-` (components)
+- `.util-` (utilities)
```

Comment:
I've never used this. Is it really universal?

Reply:
I'd probably actively argue against using this in some of the larger projects we work on. 😬

Also, we often namespace utilities as `u-` or colors as `c-`. Design systems may prefix all classes and this wouldn't work there - for example, a sparkbox design system could be `sbds-` or `sb-`.

Reply:
We are making an effort to make this universal. The argument I've heard for it is that, like you pointed out, `c-` could mean component or color or anything else. Including a couple extra characters helps to clarify what it is without needing a key or a cheatsheet. @zastrow may have something more to add here.